app/controllers/users_controller.rb

```diff
@@ -1740,6 +1740,68 @@ def bookmarks
     end
   end
 
+  USER_MENU_BOOKMARKS_LIST_LIMIT = 20
+  def user_menu_bookmarks
+    user = fetch_user_from_params
+    guardian.ensure_can_see_notifications!(user)
+    user_guardian = Guardian.new(user)
+
+    if params[:limit]
+      limit = params[:limit].to_i
+      limit = limit.clamp(1..USER_MENU_BOOKMARKS_LIST_LIMIT)
+    else
+      limit = USER_MENU_BOOKMARKS_LIST_LIMIT
+    end
+
+    reminder_notifications = user
+      .notifications
+      .visible
+      .includes(:topic)
+      .where(read: false, notification_type: Notification.types[:bookmark_reminder])
+      .limit(limit)
+      .to_a
+
+    if reminder_notifications.size < limit
+      exclude_bookmark_ids = reminder_notifications
+        .filter_map { |notification| notification.data_hash[:bookmark_id] }
```

Comment:
I think we can just use `Array#map` here since all the notifications are `bookmark_reminder` type so there will always be a `bookmark_id`.

Reply:
We've only recently started adding `bookmark_id` to the `data` hash of `bookmark_reminder` notifications (see https://github.com/discourse/discourse/pull/17547), so for now we gotta check that `bookmark_id` exist.

Reply:
Ahh icic. I think we should have backfilled though... but let's keep this as it is for now.